homeassistant/components/websocket_api/commands.py

```diff
@@ -283,18 +283,27 @@ def handle_subscribe_entities(
 ) -> None:
     """Handle subscribe entities command."""
     entity_ids = set(msg.get("entity_ids", []))
+    _cached_state_diff_message = partial(messages.cached_state_diff_message, msg["id"])
+    _send_message = connection.send_message
 
-    @callback
-    def forward_entity_changes(event: Event) -> None:
-        """Forward entity state changed events to websocket."""
-        if not connection.user.permissions.check_entity(
-            event.data["entity_id"], POLICY_READ
-        ):
-            return
-        if entity_ids and event.data["entity_id"] not in entity_ids:
-            return
+    if entity_ids or not connection.user.is_admin:
```

Comment:
With this change, what happens if a user's admin status is changed while a subscription is active?

Reply:
If their admin credentials are revoked it would not check again until the next time the stream was restarted

Reply:
But it looks like permissions policies are compiled once as well so the effect is likely the same as it currently is

Reply:
It does look like the permissions cache is invalided via `invalidate_permission_cache` so it should change midstream currently

Reply:
We should also add test coverage to ensure that guarantee is enforced as well.

Reply:
In practice the permission check is not effective security control since its not broadly enforced.  It can easily be evaded by `render_template` and many other places.

Reply:
We explicitly document it as not enabled or enforced so no vulnerability here 👍 

https://developers.home-assistant.io/docs/auth_permissions/

Reply:
So that leaves the long term question of if we plan on moving auth_permissions forward that we should not move this PR forward and rework the architecture quite a bit as the current design does not build the permissions system low enough into the stack that it would be feasible to enforce in a auditable or reliable manner and would require significant breaking changes to offer that level of guarantee (which is probably why it hasn't really moved forward).

Reply:
Unless there is going to be a concerted effort to implement it broadly low enough in the stack that it would be auditable and enforceable way, I'd be in favor of ripping out the permission checks as I wouldn't want someone to think they are an effective security control.

Reply:
That being said, the ability to limit functionality to specific users is a great feature, but it is difficult to support from a development and user perspective because of the additional complexity it adds.

It and may not be aligned well the majority of the current user base.

Reply:
Whatever the outcome of a decision to move forward the current experimental permission system is, it shouldn't be made here so let's close this PR until a future date when it can be re-examined when a decision has been made on how that will move forward